client/state/atomic-transfer-with-plugin/actions/index.ts

```diff
@@ -0,0 +1,96 @@
+import {
+	ATOMIC_PLUGIN_INSTALL_INITIATE_WITH_TRANSFER,
+	ATOMIC_PLUGIN_INSTALL_INITIATE,
+	ATOMIC_PLUGIN_INSTALL_REQUEST_TRANSFER_STATUS,
+	ATOMIC_PLUGIN_INSTALL_REQUEST_STATUS,
+	ATOMIC_PLUGIN_INSTALL_SET_TRANSFER_STATUS,
+	ATOMIC_PLUGIN_INSTALL_SET_STATUS,
+} from 'calypso/state/action-types';
+
+/**
+ * Initiate Atomic transfer with plugin install.
+ *
+ * @param {string} softwareSet Software set slug.
+ * @param {string} siteId Site ID.
+ * @returns {object} An action object.
+ */
+export const initiateAtomicTransferWithPluginInstall = (
+	softwareSet: string,
+	siteId: string
+) => ( {
+	type: ATOMIC_PLUGIN_INSTALL_INITIATE_WITH_TRANSFER,
+	siteId,
+	softwareSet,
+} );
+
+/**
+ * Initiate plugin install and activation.
+ *
+ * @todo Add additional params.
+ * @param {string} softwareSet Software set slug.
+ * @param {string} siteId Site ID.
+ * @returns {object} An action object.
+ */
+export const initiateAtomicPluginInstall = ( softwareSet: string, siteId: string ) => ( {
```

Comment:
```suggestion
export const initiateAtomicPluginInstall = ( siteId: number, softwareSet: string ) => ( {
```

It's probably best if siteId goes first for consistency with selectors, + type of siteId should be `number`.